src/main/java/com/firebolt/jdbc/metadata/FireboltDatabaseMetadata.java

```diff
@@ -175,14 +175,27 @@ public ResultSet getSchemas() throws SQLException {
 
 	@Override
 	public ResultSet getSchemas(String catalog, String schemaPattern) throws SQLException {
-		String dbName = connection.getSessionProperties().getDatabase();
-		List<String> publicRow = Arrays.asList(PUBLIC_SCHEMA_NAME, dbName);
-		List<String> informationSchemaRow = Arrays.asList(INFORMATION_SCHEMA_NAME, dbName);
-		List<String> catalogRow = Arrays.asList(CATALOG_SCHEMA_NAME, dbName);
+		String catalogClause = catalog == null ? null : format("%s LIKE '%s'", TABLE_CATALOG, catalog);
+		String schemaClause = schemaPattern == null ? null : format("TABLE_SCHEMA LIKE '%s'", schemaPattern);
+		String where = Stream.of(catalogClause, schemaClause).filter(Objects::nonNull).collect(joining(" AND "));
+		if (where.isEmpty()) {
+			where = " WHERE " + where;
+		}
```

Comment:
Should this if be an inverse instead? Otherwise if where is empty we're creating a broken SQL.